backend/src/main/java/com/empire/World.java

```diff
@@ -1902,17 +1900,12 @@ void payTroops() {
 		}
 
 		void reapHarvests() {
+			Set<String> stoicNations = new HashSet<>();
+			for (String k : kingdoms.keySet()) if (kingdoms.get(k).hasTag("Stoic")) stoicNations.add(k);
 			if (isHarvestTurn()) {
-				for (Region r : regions) if (r.isLand()) {
-					double base = r.harvest * r.population;
-					double mods = 1;
-					double unrest = r.calcUnrest(World.this);
-					if (unrest > .5 && !getNation(r.getKingdom()).hasTag("Stoic")) base *= 1.5 - unrest;
-					if (r.noble != null && r.noble.hasTag("Meticulous")) mods += .15;
-					r.food += base * mods;
-					r.harvest = harvests.get(((date + 1) / 13) % 4);
-				}
+				for (Region r : regions) r.harvest(stoicNations, World.this);
```

Comment:
Is there a reason not to just pass an isStoic flag into Region.harvest()?

Reply:
It requires the caller to look up the region's kingdom, which feels like to me like the caller having to know too much about the implementation of the function. Said another way, injecting the set of all stoic nations is cross-regional, whereas passing an isStoic means the caller has to guarantee that it understands which kingdom's stoicism is important.